src/main/kotlin/io/kuzzle/sdk/protocol/WebSocket.kt

```diff
@@ -102,11 +113,13 @@ open class WebSocket : AbstractProtocol {
                     }
                 }
             } catch (e: Exception) {
-                tryToReconnect()
+                reconnected = tryToReconnect()
+            }
+            if (! reconnected) {
+                state = ProtocolState.CLOSE
+                trigger("networkStateChange", ProtocolState.CLOSE.toString())
+                ws = null
```

Comment:
`tryToReconnect` could not be resolve soon enough before checking the `reconnected` boolean ?